src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -92,10 +93,69 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
           handler.destroy();
         });
 
+        function hasQueryOrFilters(searchSource) {
+          const filters = searchSource.get('filter');
+          if (filters.length > 0) {
+            return true;
+          }
+
+          const query = searchSource.get('query');
+          if (query.query && query.query.length > 0) {
+            return true;
+          }
+
+          return false;
+        }
+
+        function getFilterLabel(filter) {
+          if (!_.has(filter, 'meta') || (!filter.meta.alias &&
+            (!filter.meta.key && !filter.meta.value)) ) {
+            return '';
+          }
+
+          let prefix = _.get(filter, 'meta.negate', false) ? 'NOT' : '';
+          if (filter.meta.alias) {
+            return `"${prefix} ${filter.meta.alias}"`
+          }
+
+          return `"${prefix} ${filter.meta.key}:${filter.meta.value}"`;
+        }
+
+        function getSearchSourceLabel(searchSource) {
+          let label = '';
```

Comment:
You can use lodash to handle the non-existant cases, since lodash's map and get, etc work with null/undefined. Something like this would work:

```js
function getSearchSourceLabel(searchSource) {
  const filters = _(searchSource.get('filter')).map(getFilterLabel).join(',');
  const query = _(searchSource.get('query')).get('query', '');
  const filtersLabel = filters ? `filters=${filters} ` : '';
  const queryLabel = query ? `query="${query}"` : '';
  return filtersLabel + queryLabel;
}
```

Totally subjective preference, but when I see a `let`, I try to think if there's a way to express everything via `consts`.

Reply:
Similar is true for `hasQueryOrFilters` above:

```js
function hasQueryOrFilters(searchSource) {
  return searchSource.get('filter').length > 0 ||
     _(searchSource.get('query')).get('query', '').length > 0;
}
```